gcloud/orb.yaml

```diff
@@ -252,6 +296,102 @@ jobs:
           command: cat output
           when: always
 
+  deploy-gke-TODO-NAME:
+    description: >
+      Deploy a k8s deployment to GKE via provided YAML. Can optionally update
+      the autoscaler's configuration for this deployment. Will record the
+      previous deployment version in a workspace.
+    docker:
+      - image: google/cloud-sdk:alpine
+    parameters:
+      autoscaler_yaml:
+        description: >
+          The configuration YAML file for the autoscaler. If provided, will
+          attempt to update the autoscaler.
+        type: string
+        default: ""
+      cluster:
+        type: string
+      creds:
+        type: env_var_name
+      deployment:
+        description:
+          Name of your deployment. If you are using multi-deployment
+          autoscaling this may correspond to multiple actual k8s deployments.
+        type: string
+      k8s_yaml:
+        description: >
+          The k8s YAML file for your deployment(s).
+        type: string
```

Comment:
Ok so if I need to interpolate the deployment and/or autoscaler yaml you'll have to do that in a pre-step to this job?
Is that the idea?

Reply:
That's right! Different apps do this differently so I like having this step decoupled from the actual deployment.

Reply:
That's pragmatic and I like it 👍